elasticsearch/templates/deployment-es-master.yaml

```diff
@@ -0,0 +1,71 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: es-master
+  labels:
+    component: {{ .Values.labels.es_master.component }}
+    role: {{ .Values.labels.es_master.role }}
+spec:
+  replicas: {{ .Values.replicas.es_master }}
+  template:
+    metadata:
+      labels:
+        component: {{ .Values.labels.es_master.component }}
+        role: {{ .Values.labels.es_master.role }}
+      annotations:
+        pod.beta.kubernetes.io/init-containers: '[
+          {
+          "name": "sysctl",
+            "image": "busybox",
+            "imagePullPolicy": "IfNotPresent",
+            "command": ["sysctl", "-w", "vm.max_map_count=262144"],
+            "securityContext": {
+              "privileged": true
+            }
+          }
+        ]'
+    spec:
+      containers:
+      - name: es-master
+        securityContext:
+          privileged: false
+          capabilities:
+            add:
+              - IPC_LOCK
+              - SYS_RESOURCE
+        image: {{ .Values.images.elasticsearch }}
+        imagePullPolicy: {{ .Values.images.pullPolicy }}
+        env:
+        - name: NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        - name: NODE_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.name
+        - name: "CLUSTER_NAME"
+          value: "myesdb"
+        - name: "NUMBER_OF_MASTERS"
+          value: "2"
```

Comment:
These environment variables should be configurable where it makes sense.  For example here, we'd want to make sure NUMBER_OF_MASTERS matches the actual replicas of es-master pods.